src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -131,16 +132,26 @@ void CsvImportWidget::updateTableview() {
 
 void CsvImportWidget::updatePreview() {
 
+    int minSkip = 0;
+    if (m_ui->checkBoxFieldNames->isChecked())
+        minSkip = 1;
     m_ui->labelSizeRowsCols->setText(m_parserModel->getFileInfo());
-    m_ui->spinBoxSkip->setValue(0);
-    m_ui->spinBoxSkip->setMaximum(qMax(0, m_parserModel->rowCount() - 1));
+    m_ui->spinBoxSkip->setRange(minSkip, qMax(minSkip, m_parserModel->rowCount() - 1));
+    m_ui->spinBoxSkip->setValue(minSkip);
 
-    int i;
+    int i, emptyId = 0;
+    QString columnName;
     QStringList list(tr("Not present in CSV file"));
 
     for (i = 1; i < m_parserModel->getCsvCols(); ++i) {
-        QString s = QString(tr("Column ")) + QString::number(i);
-        list << s;
+        if (m_ui->checkBoxFieldNames->isChecked()) {
+            columnName = m_parserModel->getCsvTable().at(0).at(i);
```

Comment:
See comment at src/gui/csvImport/CsvParserModel.h line 53 (for some reason, link below does not point to such line)

[https://github.com/keepassxreboot/keepassxc/blob/943c3155af3355bb117263687ee16135cfa6f022/src/gui/csvImport/CsvParserModel.h#L53](url)